virtio-devices/src/iommu.rs

```diff
@@ -572,7 +572,7 @@ struct IommuEpollHandler {
     kill_evt: EventFd,
     pause_evt: EventFd,
     mapping: Arc<IommuMapping>,
-    ext_mapping: BTreeMap<u32, Arc<dyn ExternalDmaMapping>>,
+    ext_mapping: Arc<RwLock<BTreeMap<u32, Arc<dyn ExternalDmaMapping>>>>,
```

Comment:
Are you sure about `RwLock` here?

Reply:
This is a good fit for that kind of usage. We know the write lock will only be taken very infrequently while the read lock will be taken much more often.
The real good use case for it would be multiple readers but that's not the case here.
But if you want me to change it to Mutex, I'm totally fine with it.

Reply:
I agree in theory. But in practice I don't know how good it is on Linux.

Reply:
No worries, I'll change it to Mutex.

Reply:
> No worries, I'll change it to Mutex.

You shouldn't do it just because I don't know. :-) Out of interest which threads is this lock between?